### DISCUSSION TITLE
How to adapt the code for varied input feature dimensions

### DISCUSSION THREAD

**Discussion by lilytong20** [sebp/scikit-survival#450] [Q&A]
Hi, 

I have a dataset, where one input feature is a vector for each subject (i.e., more than one number per person per feature), other features are normal scaler input features (i.e., one number per subject per feature).  I'd want to try svm-, gradient boosted- and tree-based models for this data. 

However, when trying to input the single vector feature for each subject (dependent variable y is in the right format), there were errors on input dimensions or converting to python scalars, which were from some codes checking X shape. In source code, it says the input X matrix should be  n_sample x n_feature, guess this is designed for scaler features by default at the beginning?

My Q is: if it's feasible to modify the input feature X dimensions, so the svm- and tree-based models of this toolbox will still work? 

Any suggestions will be appreciated. 

Thanks a lot.

**Comment by sebp** [comment]
What is the exact error message?

Does the error occur when fitting the model, or during prediction?

**Reply by lilytong20** [reply]
Attached please find the error message. I used FastSurvivalSVM.fit(), so when fitting the model. 

For single feature from my data, how would you recommend to organise the input?  I.e., n_sample x n_feature, where each feature is a vector, for example 100 x1 vector. For more than one features, if scaler features were padded, then each subject will have a n_feature x100 dimension feature matrix, so the final input will be like: n_sample x (n_feature x 100) ndarray. 

Thanks a lot for your kind and fast reply. 

<img width="674" alt="error_adapt codes for varied input feature dimensions" src="https://github.com/sebp/scikit-survival/assets/31854140/02f9ae48-01c3-450a-8a3b-30e8a54773c8">

**Comment by sebp** [comment]
The feature matrix needs to have 2 dimension.
An input with shape n_sample x (n_feature x 100) would work.

**Reply by lilytong20** [reply]
Great to know this would work. 

Do you think n_sample x (100 x n_feature) will work too? Will the interpretations on weights and results change if the 2-D feature matrix's dimensions are transposed? I asked was for that this nice toolbox recommend n_sample x n_feature, it seems that input feature matrix is meant to arrange one feature per column. 

Appreciate a lot.

**Comment by sebp** [comment]
I'm note sure what you mean with n_sample x (100 x n_feature).
I probably should have written the shape as n_sample x (100 * n_feature)

A single sample (row) needs to be a 1D vector. I can't be a 2D matrix.

**Reply by lilytong20** [reply]
What did you mean by  (100 * n_feature) here? Thanks a lot

**Comment by sebp** [comment]
What I mean is that you need to flatten your 100 x n_features 2D matrix 
into 1D vector of length 100 * n_features.

**Reply by lilytong20** [reply]
I see. If for one sample, there's feature 1 = [x11, x12, x13], feature 2 = [x21, x22, x23], I shall input one concatenated or flattened vector as X = [x11, x12, x13, x21, x22, x23].  So a weight will be learnt for each element. 

Will try. Thank you
